### ISSUE TITLE
Multi filter of RecursiveTreeItem

### ISSUE THREAD

**Issue by luanvuhlu** [jfoenixadmin/JFoenix#658]
I created a single filter as well.
`JFXTextField filterField = new JFXTextField();
filterField.textProperty().addListener((o,oldVal,newVal)->{
    treeView.setPredicate(user -> user.getValue().age.get().contains(newVal)
                || user.getValue().department.get().contains(newVal)
                || user.getValue().userName.get().contains(newVal));
});`

However, now I have a form with multi text field, each field filters for one column and of course, all filters have to be combined. I try but it does not work

`RecursiveTreeItem<HistoryTable> historiesTableTree = new RecursiveTreeItem<>(observableList, RecursiveTreeObject::getChildren);`

`firstNameSearchField.textProperty().addListener((observable, oldValue, newValue) -> 
			treeView.getPredicate().and(user -> 
				Strings.isNullOrEmpty(newValue) || user .getValue().getFirstName().get().contains(newValue))
		);`

`lastNameSearchField.textProperty().addListener((observable, oldValue, newValue) -> 
			treeView.getPredicate().and(user -> 
				Strings.isNullOrEmpty(newValue) || user .getValue().getLastName().get().contains(newValue))
		);`

anw, your project is awesome, it's so beautiful. Thank you.

**Comment by jfoenixadmin** [comment]
Hello,
did you set the predicate? in the provided code, I noticed you created the predicate object using `and `method however you didn't set it.

Regards,

**Comment by luanvuhlu** [comment]
Thank for your help, I have found it.
Before, I thought I could combine predicate by "and" method, but like you said, I was totally wrong. So I change my code, and it's work!
Declare search value as instance variables
`String firstNameSearchValue , lastNameSearchValue ;`

`firstNameSearchField.textProperty().addListener((observable, oldValue, newValue) -> firstNameSearchValue = newValue);`

`lastNameSearchField.textProperty().addListener((observable, oldValue, newValue) -> lastNameSearchValue = newValue); `

And when I press button event:

`btnSearch.setOnAction(e -> 
			treeView.setPredicate(h ->
				 (Strings.isNullOrEmpty(firstNameSearchValue ) || h.getValue().getFirstName().get().contains(firstNameSearchValue ))
						&& (Strings.isNullOrEmpty(lastNameSearchValue) || h.getValue().getLastName().get().contains(lastNameSearchValue))
)`

However, does it have any problem if I set predicate for treeview whenever the button is clicked?
